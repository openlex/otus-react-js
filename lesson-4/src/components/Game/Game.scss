.block {
	font-family: Helvetica, Arial, sans-serif;
}

.info {
	display: flex;
	justify-content: space-between;
	align-items: center;
	margin-bottom: 10px;
}

.title {
	font-weight: bold;
}

.description {
	font-size: 14px;
	margin-bottom: 5px;
}

.genres {
	font-size: 14px;
	color: gray;
}

