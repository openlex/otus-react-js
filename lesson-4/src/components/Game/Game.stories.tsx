import * as React from "react";
import { withKnobs } from "@storybook/addon-knobs";
import { GameForm } from "@components";
import { action } from "@storybook/addon-actions";

export default {
  title: "Список игр",
  component: GameForm,
  decorators: [withKnobs],
};

export const DefaultGameForm = () => (
  <GameForm onAddGame={action("click add game")} />
);
