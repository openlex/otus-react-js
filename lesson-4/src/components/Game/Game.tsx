import React from "react";
import { GenreList, Panel } from "@components";
import * as s from "./Game.scss";
import cn from "classnames";

import { IGame } from "@types";

interface IGameProps extends IGame {
  className?: string;
}

export const Game: React.FC<IGameProps> = (props: IGameProps) => {
  const { className, title, description, genres, rating } = props;

  return (
    <Panel className={cn(s.block, className)}>
      <div className={s.info}>
        <div className={s.title} data-test-id="title">
          {title}
        </div>
        {rating && (
          <div className={s.rating} data-test-id="rating">
            {rating}
          </div>
        )}
      </div>
      {description && (
        <div className={s.description} data-test-id="description">
          {description}
        </div>
      )}
      {genres && (
        <div className={s.genres} data-test-id="genres-wrapper">
          <GenreList genres={genres} />
        </div>
      )}
    </Panel>
  );
};
