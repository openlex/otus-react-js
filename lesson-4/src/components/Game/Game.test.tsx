import React from "react";
import { Game } from "./Game";
import { render, shallow } from "enzyme";

describe("Game item markup", () => {
	const props = {
		id: '1',
		title: 'Title',
	}
	const defaultGame = shallow(<Game {...props}/>);

	it("check title content is the same as in props", () => {
		expect(defaultGame.find('[data-test-id="title"]').text()).toEqual(props.title)
	});

	it("has rating, if is set", () => {
		defaultGame.setProps({
			rating: 5,
		})

		expect(defaultGame.find('[data-test-id="rating"]').length).toBe(1);
	});

	it("has no rating, if it is not set", () => {
		defaultGame.setProps({
			rating: undefined,
		})

		expect(defaultGame.find('[data-test-id="rating"]').length).toBe(0);
	});

	it("has no rating, if it is set 0", () => {
		defaultGame.setProps({
			rating: 0,
		})

		expect(defaultGame.find('[data-test-id="rating"]').length).toBe(0);
	});

	it("rating is the same as in props", () => {
		const rating = 5;
		const game = render(<Game {...props} rating={rating}/>);

		expect(game.find('[data-test-id="rating"]').text()).toEqual(rating.toString())
	});

	it("has description, if is set", () => {
		const game = render(<Game {...props} description="Desc"/>);

		expect(game.find('[data-test-id="description"]').length).toBe(1);
	});

	it("Has no description, if it is not set", () => {
		expect(defaultGame.find('[data-test-id="description"]').length).toBe(0);
	});

	it("has genres if it exist", () => {
		const game = shallow(<Game {...props} genres={['1','2','3']}/>);

		expect(game.find('[data-test-id="genres-wrapper"]').length).toBe(1);
	});

	it("Match Snapshot", () => {
		const game = shallow(<Game {...props} rating={4} description="Desc" genres={['1','2','3']}/>);

		expect(game).toMatchSnapshot();
	});

})
