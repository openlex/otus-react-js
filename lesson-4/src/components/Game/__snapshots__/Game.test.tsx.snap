// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game item markup Match Snapshot 1`] = `ShallowWrapper {}`;
